nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,642 @@
+            self.cv_params_.setdefault(c, {})
+            for k in best_params:
+                self.cv_params_[c].setdefault(k, []).append(best_params[k])
+            cv_pred.setdefault(c, []).append(best_y['y_pred'])
+            cv_true.setdefault(c, []).append(best_y['y_true'])
+
+            if self.is_binary_:
+                other_class = np.setdiff1d(
+                    self.classes_, classes_to_predict)[0]
+                cv_pred.setdefault(other_class, []).append(best_y['inverse'])
+                coefs.setdefault(other_class, []).append(-best_coef)
+                intercepts.setdefault(other_class, []).append(-best_intercept)
+
+        # Transforming cv_pred into a list of dict
+        y_prob = [{k: v[i] for k, v in cv_pred.items()}
+                  for i in range(len(cv_pred[cv_pred.keys()[0]]))]
+
+        self.cv_y_pred_ = []
+        self.cv_y_true_ = []
+        self.cv_scores_ = []
+
+        if is_classification_:
+            classes = self.classes_
+            for k in range(len(y_prob)):
+                y_prob_ = np.vstack([y_prob[k][c] for c in classes]).T
+                y_pred_ = self.classes_[np.argmax(y_prob_, axis=1)]
+                y_true_ = cv_true[cv_true.keys()[0]][k]
+
+                self.cv_y_pred_.append(y_pred_)
+                self.cv_y_true_.append(y_true_)
+                self.cv_scores_.append(scorer._score_func(y_true_, y_pred_))
+        else:
+            classes = classes_to_predict
+            for k in range(len(y_prob)):
+                y_pred_ = np.vstack([y_prob[k][c] for c in classes]).T
+                y_true_ = cv_true[cv_true.keys()[0]][k]
+
+                self.cv_y_pred_.append(y_pred_)
+                self.cv_y_true_.append(y_true_)
+                self.cv_scores_.append(scorer._score_func(y_true_, y_pred_))
+
+            self.cv_params_['beta'] = self.cv_params_.pop(None)
+
+        self.coef_ = np.vstack([np.mean(coefs[c], axis=0) for c in classes])
+        self.std_coef_ = np.vstack([np.std(coefs[c], axis=0) for c in classes])
+        self.intercept_ = np.hstack([np.mean(intercepts[c], axis=0)
+                                     for c in classes])
+        coef_img = {}
+        std_coef_img = {}
+        for c, coef, std in zip(classes, self.coef_, self.std_coef_):
+            c = 'beta' if c is None else c
+            coef_img[c] = self.masker_.inverse_transform(coef)
+            std_coef_img[c] = self.masker_.inverse_transform(std)
+
+        self.coef_img_ = coef_img
+        self.std_coef_img_ = std_coef_img
+
+    def decision_function(self, niimgs, index=None):
+        """Provide prediction values for new X which can be turned into
+        a label by thresholding
+        """
+        if index is not None and isinstance(index, collections.Iterable):
+            niimgs = index_img(niimgs, index)
+
+        X = self.masker_.transform(niimgs)
+        if isinstance(X, tuple):
+            X = np.vstack(X)
+        X_view = X.view()
```

Comment:
Does view() on numpy arrays really make sense if you do not specify a non-identical dtype?